fastify.js

```diff
@@ -792,6 +792,10 @@ function fastify (options) {
     let isAsync
     return function preRouting (req, res) {
       // only call isAsyncConstraint once
+      /* istanbul ignore next */
+      if (req.originalUrl == null) {
+        req.originalUrl = req.url
+      }
```

Comment:
You should place this block inside the `if (rewriteUrl)` one

Reply:
I don't think that would be a good idea because it would be better if we always give access to `originalUrl` even when not redirected. Inherently, `req.originalUrl` is not set because at each request we do `req.originalUrl = req.originalUrl || req.url` which takes the old one, if not null, else the `req.url`.